pkg/ottl/ottlfuncs/func_merge_maps.go

```diff
@@ -20,8 +20,9 @@ const (
 
 type MergeMapsArguments[K any] struct {
 	Target   ottl.PMapGetter[K]
-	Source   ottl.PMapGetter[K]
 	Strategy string
```

Comment:
Since we're looking at breaking this, should we make the strategy optional and default to "upsert"? It would be nice if we could keep `source` as the second argument and just append "sources" to the end.

If we kept `sources` as the last argument, we could even avoid a breaking change here.

Reply:
to be clear, are you suggesting:

```
type MergeMapsArguments[K any] struct {
	Target   ottl.PMapGetter[K]
	Source   ottl.PMapGetter[K]
	Strategy ottl.Optionalstring] - default to upsert
	Sources  ottl.Optional[ottl.PMapSliceGetter[K]] - default to empty
}
```

Thanks!

Reply:
>Why don't then keep the Strategy as it is and the only change would be adding an optional sources parameter at the end?

If we do that, then we can't make `Source` optional and we would need to figure out how to reconcile both `Source` and `Sources` being set.

Reply:
Ok, so this is the expected state?

```
type MergeMapsArguments[K any] struct {
	Target   ottl.PMapGetter[K]
	Source   ottl.Optional[ottl.PMapGetter[K]]
	Strategy ottl.Optional[string] - default to upsert
	Sources  ottl.Optional[ottl.PMapSliceGetter[K]] - default to empty
}
```

That means, all of the parameters except the target are optional?

Reply:
Yes, that looks right. One of either `Source` or `Sources` would be required though.

Reply:
adapted, should be ready for a re-review